### PR TITLE
xplugd: init at 1.4

### PR DESCRIPTION
**Pull request by akho**
###### Motivation for this change

[xplugd](https://github.com/troglobit/xplugd) not currently packaged.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/X11/xplugd/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "xplugd";
+  version = "1.4";
+
+  src = fetchTarball {
+    url =
+      "https://github.com/troglobit/xplugd/releases/download/v1.4/xplugd-1.4.tar.xz";
+    sha256 = "1hcj5pains83gb11kr4vm58i4mfrgw7zcjh96258i3nq79xcyw93";
+  };
```
Comment (Pacman99) [inline pkgs/tools/X11/xplugd/default.nix:15]: Can you use fetchFromGitHub instead? That would be preferred.
```suggestion
  src = fetchFromGitHub {
    owner = "troglobit";
    repo = "xplugd";
    rev = "v${version}";
    sha256 = "";
  };
```

Comment (akho) [reply pkgs/tools/X11/xplugd/default.nix]: Thank you for your suggestion, I've amended my PR to incorporate. Resolving

**Comment by ckiee** [comment]
Result of `nixpkgs-review pr 131368` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>xplugd</li>
  </ul>
</details>

File: pkgs/tools/X11/xplugd/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, stdenv
+, xorg
+, pkgconfig
```
Comment (Artturin) [inline pkgs/tools/X11/xplugd/default.nix:4]: aliases aren't allowed in nixpkgs
```suggestion
, pkg-config
```

File: pkgs/tools/X11/xplugd/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "troglobit";
+    repo = "xplugd";
+    rev = "v${version}";
+    sha256 = "11vjr69prrs4ir9c267zwq4g9liipzrqi0kmw1zg95dbn7r7zmql";
+  };
+
+  nativeBuildInputs = with xorg; [ libX11 libXi libXrandr libXext ];
+  buildInputs = [ pkgconfig autoreconfHook ];
```
Comment (Artturin) [inline pkgs/tools/X11/xplugd/default.nix:21]:
```suggestion
  buildInputs = [ pkg-config autoreconfHook ];
```

**Review by Artturin — CHANGES_REQUESTED** [review]

File: pkgs/tools/X11/xplugd/default.nix (hunk 2 above)
Comment (akho) [reply pkgs/tools/X11/xplugd/default.nix]: Thank you, done

File: pkgs/tools/X11/xplugd/default.nix (hunk 3 above)
Comment (akho) [reply pkgs/tools/X11/xplugd/default.nix]: Thank you, done

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -28289,6 +28289,8 @@ in
 
   xpdf = libsForQt5.callPackage ../applications/misc/xpdf { };
 
+  xplugd = callPackage ../tools/X11/xplugd {};
```
Comment (ckiee) [inline pkgs/top-level/all-packages.nix:28292]:
```suggestion
  xplugd = callPackage ../tools/X11/xplugd { };
```

**Review by ckiee — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (ckiee) [reply pkgs/top-level/all-packages.nix:28292]: By the way, have a read of the comment at the top:
```nix
/* The top-level package collection of nixpkgs.
 * It is sorted by categories corresponding to the folder names
 * in the /pkgs folder. Inside the categories packages are roughly
 * sorted by alphabet, but strict sorting has been long lost due
 * to merges. Please use the full-text search of your editor. ;)
 * Hint: ### starts category names.
 */
```

Comment (akho) [reply pkgs/top-level/all-packages.nix:28292]: Yeah, I’ve inserted before xpointerbarrier, which follows alphabetically in tools/X11. Thanks for the suggestion.

Comment (ckiee) [reply pkgs/top-level/all-packages.nix:28292]: Yeah, I meant that they are in clusters but they aren't in the correct categories (see the line with `### TOOLS`)

File: pkgs/tools/X11/xplugd/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "troglobit";
+    repo = "xplugd";
+    rev = "v${version}";
+    sha256 = "11vjr69prrs4ir9c267zwq4g9liipzrqi0kmw1zg95dbn7r7zmql";
+  };
+
+  nativeBuildInputs = with xorg; [ libX11 libXi libXrandr libXext ];
+  buildInputs = [ pkg-config autoreconfHook ];
```
Comment (Artturin) [inline pkgs/tools/X11/xplugd/default.nix:21]:
```suggestion
  nativeBuildInputs = [ pkg-config autoreconfHook ];
  buildInputs = with xorg; [ libX11 libXi libXrandr libXext ];
```

Comment (akho) [reply pkgs/tools/X11/xplugd/default.nix]: Thank you, done

**Review by Artturin — APPROVED** [review]

**Pull request merged**